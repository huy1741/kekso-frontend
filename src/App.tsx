import React, { useEffect, useState } from 'react';
import './index.css';
import { ItemList } from './ItemListComp/ItemList';
import { SearchBar } from './SearchBarComp/SearchBar';
import apiUtils from './services/api';
import orderInterface from './type';

function App() {
  const [checkedBox, setCheckedBox] = useState(true);
  const [orderList, changeOrderList] = useState<orderInterface[]>([])
  const [itemName, changeItemName] = useState('')
  const [isLoading, setLoading] = useState(true)
  const [isError, setError] = useState(false)
  const fetchOrders = () => {
    apiUtils.getData()
      .then(function (response) {
        changeOrderList(response.filterData)
      })
      .catch(function (error) {
        setError(true)
      })
      .finally(function () {
        setLoading(false)
      });
  }
  const handleTextChange = (text: string) => {
    changeItemName(text)
  }

  const handleCheckBox = () => {
    setCheckedBox(!checkedBox)
  }
  //return array based on search term or checkbox
  const renderFilteredArray = (): orderInterface[] => {
    let array = orderList
    if (checkedBox === true) {
      array = array.filter(item => item.ShippedDate !== null)
    }
    if (itemName === '') {
      return array
    }
    else {
      return array.filter(item => JSON.stringify(item.ProductList).toLowerCase().includes(itemName.toLowerCase()))
    }
  }
  useEffect(() => {
    // Fetch all orders
    fetchOrders()
  }, []);
  return (
    <div>
      <div>
        <p>NORTHWIND</p>
      </div>
      {isLoading ?
        <div className="loader"></div> :
        <>
          <SearchBar
            checkedBox={checkedBox}
            onBoxClicked={handleCheckBox}
            handleTextChange={handleTextChange} />
          <div className="listTable">
            {renderFilteredArray().length > 0 ? renderFilteredArray().map((item, index) => <ItemList
              hashNum={index} orderListDetails={item} key={index}
            />) : <p className='warnMessage'>There is no item fit your search, please try again.</p>}
          </div>
        </>}
      {isError && <p className='warnMessage'>Something went wrong, please try again.</p>}
    </div>
  );
}

export default App;
