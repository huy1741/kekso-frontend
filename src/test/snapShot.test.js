import renderer from "react-test-renderer";
import App from "../App";
import { act } from "react-dom/test-utils";
it("renders the app correctly", async () => {
  let component;
  await act(async () => {
    component = await renderer.create(<App />);
  });
  await expect(component.toJSON()).toMatchSnapshot();
});
