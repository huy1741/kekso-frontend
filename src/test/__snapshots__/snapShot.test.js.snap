// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app correctly 1`] = `
<div>
  <div>
    <p>
      NORTHWIND
    </p>
  </div>
  <div
    className="loader"
  />
</div>
`;
